core/src/main/scala/org/apache/spark/rpc/akka/AkkaRpcEnv.scala

```diff
@@ -212,7 +211,7 @@ private[spark] class AkkaRpcEnv private[akka] (
 
   override def asyncSetupEndpointRefByURI(uri: String): Future[RpcEndpointRef] = {
     import actorSystem.dispatcher
-    actorSystem.actorSelection(uri).resolveOne(defaultLookupTimeout).
+    actorSystem.actorSelection(uri).resolveOne(defaultLookupTimeout.duration).
```

Comment:
what you are describing sounds reasonable.  We certainly don't want to change the return type or await on the result here.

You could potentially fix the doubly-appended msg by throwing a subclass of `TimeoutException`, which then you could check for before you amend a msg.  However, this is really begging the question -- if there is one timeout passed in here, and another is passed to an `Await.result` later, how is akka using them both?  I suppose it will timeout on whichever one is shorter?  I don't see anything very definitive in the docs, perhaps we should confirm on akka-user.  That would inform how we should do the error handling.

(Honestly I thiink it will also be fine to not stress too much about this case, it may not be worth it.)